frontends/web/src/hooks/qrcodescanner.test.ts

```diff
@@ -13,37 +13,40 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+import { afterEach, describe, expect, it, Mock, vi } from 'vitest';
 import { RefObject } from 'react';
-import { act, renderHook } from '@testing-library/react-hooks';
+import { act, renderHook, waitFor } from '@testing-library/react';
 import QrScanner from 'qr-scanner';
 import { useHasCamera, useQRScanner } from './qrcodescanner';
 
-
 const mockScanner = {
-  start: jest.fn(() => Promise.resolve()),
-  stop: jest.fn(),
-  destroy: jest.fn(),
-  hasCamera: jest.fn(() => Promise.resolve()),
+  start: vi.fn(() => Promise.resolve()),
+  stop: vi.fn(),
+  destroy: vi.fn(),
+  hasCamera: vi.fn(() => Promise.resolve()),
 };
 
-jest.mock('qr-scanner', () => {
+vi.mock('qr-scanner', () => {
   class MockedQrScanner {
     start = mockScanner.start();
     stop = mockScanner.stop();
     destroy = mockScanner.destroy();
-    static hasCamera: jest.Mock<Promise<void>, []>;
+    // TODO: I don't know how to migrate this
+    // @ts-ignore
+    static hasCamera: vi.Mock<Promise<void>, []>;
```

Comment:
this is blocking this PR  and the only issue that I am aware of. I am thinking about just nuking the whole test.

Reply:
ack. I'll remove the test. I feel a bit sorry to kill it 🙈